dashboard/app/models/pd/teacher_application.rb

```diff
@@ -88,15 +94,27 @@ def validate_required_application_fields
 
   validate :primary_email_must_match_user_email
   def primary_email_must_match_user_email
-    if user && user.hashed_email != Digest::MD5.hexdigest(primary_email)
-      account_settings_link = ActionController::Base.helpers.link_to('account settings', '/users/edit')
-      errors.add(
-        :primary_email,
-        "must match your login. If you want to use this email instead, first update it in #{account_settings_link}."
+    return unless user
```

Comment:
It's not. `primary_email_matches_user_email?` will return true only if there is a user and its email matches, meaning it will return false if there is no user. Here I specifically don't want to fail the validation if there is no user.